app/client/cli/servicer.go

```diff
@@ -0,0 +1,284 @@
+	}
+
+	var (
+		servicer rpc.ProtocolActor
+		found    bool
+	)
+	// TODO: a map may be a better choice for storing servicers
+	for _, s := range session.Servicers {
+		if s.Address == servicerAddress {
+			servicer = s
+			found = true
+			break
+		}
+	}
+
+	// TODO: cover with unit tests
+	if !found {
+		return rpc.Session{}, rpc.ProtocolActor{}, fmt.Errorf("Error getting servicer: address %s does not match any servicers in the session", servicerAddress)
+	}
+
+	// TODO: cover with unit tests
+	found = false
+	for _, ch := range servicer.Chains {
+		if ch == chain {
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		return rpc.Session{}, rpc.ProtocolActor{}, fmt.Errorf("Error getting servicer: service %s does not support chain %s", servicerAddress, chain)
+	}
+
+	return *session, servicer, nil
+}
+
+func getCurrentSession(ctx context.Context, appAddress, chain string) (*rpc.Session, error) {
+	// TODO: passing 0 as the height value to get the current session seems more optimal than this.
+	currentHeight, err := getCurrentHeight(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("Error getting current session: %w", err)
+	}
+
+	req := rpc.SessionRequest{
+		AppAddress: appAddress,
+		Chain:      chain,
+		// TODO: Geozone
+		SessionHeight: currentHeight,
+	}
+
+	client, err := rpc.NewClientWithResponses(remoteCLIURL)
+	if err != nil {
+		return nil, fmt.Errorf("Error getting current session for app/chain/height: %s/%s/%d: %w", appAddress, chain, currentHeight, err)
+	}
+
+	resp, err := client.PostV1ClientGetSessionWithResponse(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("Error getting current session with request %v: %w", req, err)
+	}
+	// TODO: refactor boiler-plate code
+	if resp.HTTPResponse.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("Error getting current session: Unexpected status code %d for request %v", resp.HTTPResponse.StatusCode, req)
+	}
+
+	if resp.JSON200 == nil {
+		return nil, fmt.Errorf("Error getting current session: Unexpected response %v", resp)
+	}
+
+	return resp.JSON200, nil
+}
+
+// TODO: reuse this function in the query commands
+func getCurrentHeight(ctx context.Context) (int64, error) {
+	client, err := rpc.NewClientWithResponses(remoteCLIURL)
+	if err != nil {
+		return 0, fmt.Errorf("Error getting current height: %w", err)
+	}
+	resp, err := client.GetV1QueryHeightWithResponse(ctx)
+	if err != nil {
+		return 0, fmt.Errorf("Error getting current height: %w", err)
+	}
+	if resp.HTTPResponse.StatusCode != http.StatusOK {
+		return 0, fmt.Errorf("Error getting current height: Unexpected status code %d", resp.HTTPResponse.StatusCode)
+	}
+
+	if resp.JSON200 == nil {
+		return 0, fmt.Errorf("Error getting current height: Unexpected response %v", resp)
+	}
+
+	return resp.JSON200.Height, nil
+}
+
+// TODO: (localnet) Publish Servicer(s) Host and Port as env. vars in K8s: similar to Validators
+// TODO: (REFACTOR) should we move package-level variables (e.g. remoteCLIURL) to a cli object?
```

Comment:
We have a `REFACTOR` tag (see Makefile). +1 to the refactor

cc @h5law for the idea

Reply:
Thanks. Fixed.